lib/udf_json_dicts.rb

```diff
@@ -0,0 +1,46 @@
+class UdfJsonDicts
+  UDFS = [
+      {
+          type:        :function,
+          name:        :json_extract_path_keys,
+          description: "Return the list of keys in a json dict as a json string",
+          params:      "jsonstr varchar(max)",
+          return_type: "varchar(max)",
+          body:        %~
+            import json
+            if not jsonstr:
+              return '[]'
+            else:
+              return json.dumps(sorted(json.loads(str(jsonstr)).keys()))
```

Comment:
Not sure what you mean. Do you mean that this operation would be _too heavy_ to have as a scalar UDF function in Redshift? Well, yes this is is quite heavy operation but no way of implementing this in Redshift except with an UDF. 

The main reason for the sorting is to have a deterministic behaviour for the function when you try to extract *at most* K keys from a dictionary. This is needed when processing A/B experiments metadata in our ETL where we need to put an upper limit in the number of experiments that apply in a page view:
https://github.com/FindHotel/etl/blob/master/etl/redshift/_job/nlucro-stage5/10/load_nlucro.nfh_experiment.sql#L29